web/src/utilities/utilities.js

```diff
@@ -672,11 +672,25 @@ export const Utilities = {
     }
   },
 
-  isClusterUpgrading(state) {
-    const normalizedState = this.clusterState(state);
-    return (
-      normalizedState === "Upgrading" || normalizedState === "Upgrading addons"
-    );
+  isClusterUpgrading(cluster) {
+    if (!cluster) {
+      return false;
+    }
+    const normalizedState = this.clusterState(cluster.state);
+    if (
+      normalizedState === "Upgrading" ||
+      normalizedState === "Upgrading addons"
+    ) {
+      return true;
+    }
+
+    if (cluster.numInstallations > 1 && !cluster.state) {
```

Comment:
Is it possible to have no state with a single installation (i.e. numInstallations equals 1)? If not, does returning the number of installations matter, or can we simply check if there's no state?

Reply:
potentially during an initial install right after the first installation is created (though unlikely to impact the UI). knowing that it's not the first installation lets us know that it's an update vs install.

Reply:
Will the UI ever be available/accessible when the initial installation does not have a state yet?

Reply:
it seems very unlikely (maybe impossible), but the goal is to guard this conditional and prevent showing UI elements related to upgrades during an initial install flow.